### DISCUSSION TITLE
Variable-width layouts

### DISCUSSION THREAD

**Discussion by spencermountain** [vadimdemedes/ink#493] [Q&A]
hello, apologies if this is foolish or out-of-scope -  is there any way to mitigate ink layout issues in non-monospace situations?

I've got a script that looks great in the terminal, then gets bunged-up when pasted elsewhere. It would be cool if there was some way to anticipate this.

I don't know enough about the subject - are there some zany unicode whitespaces that take up a full fixed-width?

![image](https://user-images.githubusercontent.com/399657/146037358-d7d90fab-4d32-431a-a33c-c2bfec5d4344.png)

![image](https://user-images.githubusercontent.com/399657/146037388-63507c26-dcc9-4e74-a971-b973c8305425.png)

thanks!

**Comment by vadimdemedes** [comment]
I thought a space takes the same width as any character in monospace fonts?

**Reply by spencermountain** [reply]
yeah, that's right. 
I'm trying to get an Ink layout looking sensible in a variable-width setting. 
The (bad) idea was to support Ink-based diagrams in comments, emails, slack, etc. 
Are there any clever ways to attempt this? Thought I'd ask the pro.
thanks

**Reply by vadimdemedes** [reply]
Oh, I see. I don't think that's going to work, because Ink's entire layout system depends on a monospace font being used (as they usually are in terminals).
